File: backend/quote-service/src/main/java/com/easyerp/quoteservice/requests/QuoteRequest.java
package com.easyerp.quoteservice.requests;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@JsonIgnoreProperties(ignoreUnknown = true)
public class QuoteRequest {
    private Long clientId;
    private Double tva;
    private boolean draft;
    private List<QuoteLineRequest> lines;
}

File: client/src/models/ApiRequests/QuoteRequest.ts
import { QuoteLineRequest } from "./QuoteLineRequest";

export class QuoteRequest {
  clientId?: number;
  tva: number;
  draft: boolean;
  lines: QuoteLineRequest[];

  constructor() {
    this.tva = 0;
    this.draft = true;
    this.lines = [];
  }
}
